src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/persistence/PersistenceMapKeyDiagnosticsCollector.java

```diff
@@ -37,81 +37,56 @@ protected String getDiagnosticSource() {
-                                PersistenceConstants.DIAGNOSTIC_CODE_INVALID_ANNOTATION, null,
-                                DiagnosticSeverity.Error));
-                    }
-                    // If we have multiple MapKeyJoinColumn annotations on a single method we must
-                    // ensure each has a name and referencedColumnName
-                    if (mapKeyJoinCols.size() > 1) {
-                        validateMapKeyJoinColumnAnnotations(mapKeyJoinCols, method, unit, diagnostics);
-                    }
+                    collectDiagnostics(unit, diagnostics, type, method);
                 }
-
                 // Go through each field to ensure they do not have both MapKey and MapKeyColumn
                 // Annotations
                 fields = type.getFields();
                 for (PsiField field : fields) {
-                    List<PsiAnnotation> mapKeyJoinCols = new ArrayList<PsiAnnotation>();
-                    boolean hasMapKeyAnnotation = false;
-                    boolean hasMapKeyClassAnnotation = false;
-                    allAnnotations = field.getAnnotations();
-                    for (PsiAnnotation annotation : allAnnotations) {
-                        String matchedAnnotation = getMatchedJavaElementName(type, annotation.getQualifiedName(),
-                                PersistenceConstants.SET_OF_PERSISTENCE_ANNOTATIONS);
-                        if (matchedAnnotation != null) {
-                            if (PersistenceConstants.MAPKEY.equals(matchedAnnotation))
-                                hasMapKeyAnnotation = true;
-                            else if (PersistenceConstants.MAPKEYCLASS.equals(matchedAnnotation))
-                                hasMapKeyClassAnnotation = true;
-                            else if (PersistenceConstants.MAPKEYJOINCOLUMN.equals(matchedAnnotation)) {
-                                mapKeyJoinCols.add(annotation);
-                            }
-                        }
-                    }
-                    if (hasMapKeyAnnotation && hasMapKeyClassAnnotation) {
-                        // A single field cannot have the same
-                        diagnostics.add(createDiagnostic(field, unit,
-                                Messages.getMessage("MapKeyAnnotationsNotOnSameField"),
-                                PersistenceConstants.DIAGNOSTIC_CODE_INVALID_ANNOTATION, null,
-                                DiagnosticSeverity.Error));
-                    }
-                    if (mapKeyJoinCols.size() > 1) {
-                        validateMapKeyJoinColumnAnnotations(mapKeyJoinCols, field, unit, diagnostics);
-                    }
+                    collectDiagnostics(unit, diagnostics, type, field);
+                }
+            }
+        }
+    }
+
+    private void collectDiagnostics(PsiJavaFile unit, List<Diagnostic> diagnostics,
+                                    PsiClass type, PsiJvmModifiersOwner fieldOrProperty) {
```

Comment:
Looking at the two spots where this method is called, would `fieldOrMethod` make more sense for the last param name than `fieldOrProperty`?

Reply:
See my response to the other comment. It's the correct terminology for the type of method. Entities can use either field or property based access.